modules/stampcollection.py

```diff
@@ -1,12 +1,28 @@
+from enum import Enum
 import re
+from typing import DefaultDict, Union
 import discord
 import numpy as np
 from utilities import utilities
 from modules.module import Module, Response
 from config import rob_id, god_id, stampy_id
 from config import stamp_scores_csv_file_path, Services
 from utilities.discordutils import DiscordMessage
-
+from collections import defaultdict
+
+class StampType(Enum):
+    GOLD = 0
+    RED = 1
+    OTHER = 2
+    
+# For any string that is neither "stamp" nor "goldstamp", returns `StampType.OTHER`
+stamp_type_str2enum: DefaultDict[str, StampType] = defaultdict(
+    lambda: StampType.OTHER,
+    **{
+        "stamp": StampType.RED,
+        "goldstamp": StampType.GOLD
+    }
+    )
```

Comment:
IMHO these enums feel overcomplicated, there are 2 lookups: from emoji to enum, then from enum to vote_strength,

I would be more comfortable with just 1 lookup - fewer moving parts that can go wrong, only 1 place in the code to update when needed:
```py
vote_strengths_per_emoji = {
 "stamp": 1,
 "goldstamp": 5
}
```
and then:
```py
vote_strength = vote_strengths_per_emoji.get(emoji, 0)
```

Reply:
I'm inclined to agree. I assumed that @UlisseMini meant to refactor strings in cases like these into `Enum`s. I'm a newbie to Stampy but I'm not sure in what cases should `vote_strength` be `0`? Intuitively, this doesn't make much sense, so maybe we could just change it into a boolean, `is_gold_stamp`?

Reply:
Couldn't we just have the strength be a built in property of the Enum?

```py
@property
def vote_strength(self) -> int:
    return self.__value__
```

You would need to make sure to set the proper values in the declaration though.

You could also do

```py
def __int__(self) -> int:
    return self.vote_strength()
```

for an easy alternative option.

Reply:
@tayler6000 Both of these throw errors

Reply:
there are thousands of emojis for Discord reactions (and whatever they call it on other supported clients/services):
![image](https://user-images.githubusercontent.com/1087670/190892998-196708fa-346a-4f8e-ba1c-c4176179d47d.png)

out of those, 2 have a special meaning: `stamp` provides a vote and `goldstamp`  provides 5x more more votes .. I still believe my suggested code is the best match for that.

Reply:
I agree but IMO calling the argument `stamp_type` when it can be any emoji is confusing. I suggest either changing it to `emoji` or `reaction` and then doing it the way suggested or checking whether the emoji is a stamp before calling `update_vote`.

Reply:
oh, right, if it wasn't clear from my suggestion, I'm happy to rename the argument to `emoji` (or `reaction` if that will look better in your opinion)

Reply:
I think I see now why it was done this way. `reset_stamps` calls `update_vote` with `stamp_type=god_id="0"` in order to recalculate stamps and do other updates. Maybe we should remove the part that does that (starting from `self.utils.users = self.utils.get_users()` AFAIK) to a separate method in order to keep one responsibility per method?